manta-crypto/src/algebra.rs

```diff
@@ -308,3 +441,37 @@ pub mod security {
     /// ```
     pub trait DecisionalDiffieHellmanHardness: ComputationalDiffieHellmanHardness {}
 }
+
+/// Testing Framework
+#[cfg(feature = "test")]
+#[cfg_attr(doc_cfg, doc(cfg(feature = "test")))]
+pub mod test {
+    use super::*;
+    use crate::eclair::{bool::Assert, cmp::PartialEq};
+
+    /// Tests if windowed scalar multiplication of the bit decomposition of `scalar` with `point` returns the
+    /// product `scalar` * `point`
+    #[inline]
+    pub fn window_correctness<S, G, F, B, COM>(
+        window_size: usize,
+        scalar: &S,
+        point: G,
+        bit_conversion: F,
+        compiler: &mut COM,
+    ) where
+        G: Clone
+            + ConditionalSelect<COM>
+            + PartialEq<G, COM>
+            + ScalarMulGroup<S, COM, Output = G>
+            + Zero<COM>,
+        F: FnOnce(&S, &mut COM) -> B,
+        B: IntoIterator<Item = Bool<COM>>,
+        COM: Assert,
+    {
+        let product = point.scalar_mul(scalar, compiler);
+        let window = Window::new(window_size, point, compiler);
+        let bits = Vec::from_iter(bit_conversion(scalar, compiler)); // TODO: use iter instead of Vec.
+        let windowed_product = window.scalar_mul(&bits, compiler);
+        product.assert_equal(&windowed_product, compiler);
```

Comment:
We might also need to check if all constraints are satisfied here?

Reply:
Yes, you're right. However, I decided not to include it here because it'd be part of a separate PR I'm working on https://github.com/Manta-Network/manta-rs/pull/208, where I'll have to think about a nice framework for circuit constraint measurement and so on. 

Even if it didn't make it to the code for this very reason, I carried out a test in `manta-pay::ecc::arkworks.rs` imitating yours and cs.is_satisfied() turned true and the constraint measurement was the mathematically expected one (around 500 constraints in savings with a window of size 2).